services/src/main/java/org/keycloak/vault/PlainTextVaultProvider.java

```diff
@@ -0,0 +1,62 @@
+package org.keycloak.vault;
+
+import org.jboss.logging.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.invoke.MethodHandles;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * A text-based vault provider, which stores each secret in a separate file. The file name needs to match a
+ * vault secret id (or a key for short). A typical vault directory layout looks like this:
+ * <pre>
+ *     ${VAULT}/key1 (contains secret for key 1)
+ *     ${VAULT}/key2 (contains secret for key 2)
+ *     etc...
+ * </pre>
+ * This kind of layout is used by Kubernetes by default (when mounting a volume into the pod).
+ * See https://kubernetes.io/docs/concepts/configuration/secret/
+ *
+ * @author Sebastian Łaskawiec
+ */
+public class PlainTextVaultProvider implements VaultProvider {
+
+    private static final Logger logger = Logger.getLogger(MethodHandles.lookup().lookupClass());
+
+    private final Path vaultPath;
+
+    public PlainTextVaultProvider(Path path) {
+        vaultPath = path;
+        logger.debugf("PlainTextVaultProvider will operate in %s directory", vaultPath.toAbsolutePath().toString());
+    }
+
+    @Override
+    public VaultRawSecret obtainSecret(String vaultSecretId) {
+        Path secretPath = vaultPath.resolve(Paths.get(vaultSecretId));
+        if (!Files.exists(secretPath)) {
+            //FIXME: I'm not sure if this distinguishes empty and non-existing secret enough.
+            //       Maybe, VaultProvider should have the following signature:
+            //       Optional<VaultRawSecret> obtainSecret(String vaultSecretId);
+            //       Or another idea is that VaultRawSecret would have a methods:
+            //       - #isEmpty() - true if the secret was created from byte[0]
+            //       - #exists() - true if the secret simply doesn't exist
+            return DefaultVaultRawSecret.empty();
```

Comment:
Not following here. Empty means that the secret has not been found. Empty secret is empty byte array (array of length 0)

Reply:
Ok, sorry my bad. I misinterpreted the API. Now it should be better.